src/combinatorics/simple_graph/connectivity.lean

```diff
@@ -1118,19 +1128,91 @@ map_injective f.injective u v
 
 end path
 
+/-! ### Transferring between graphs -/
+
+namespace walk
+
+variables {G}
+
+/-- The walk `p` transferred to lie in `H` -/
+@[protected, simp] def transfer : Π {u v : V} (p : G.walk u v) (H : simple_graph V)
+  (h : ∀ e, e ∈ p.edges → e ∈ H.edge_set), H.walk u v
+| _ _ (walk.nil) H h := walk.nil
+| _ _ (walk.cons' u v w a p) H h :=
+  walk.cons (h (⟦(u, v)⟧ : sym2 V) (by simp)) (p.transfer H (λ e he, h e (by simp [he])))
+
+variables {u v w : V} (p : G.walk u v) (q : G.walk v w)
+  {H : simple_graph V}
+  (hp : ∀ e, e ∈ p.edges → e ∈ H.edge_set)
+  (hq : ∀ e, e ∈ q.edges → e ∈ H.edge_set)
+
+lemma transfer_self : p.transfer G p.edges_subset_edge_set = p :=
+by { induction p; simp only [*, transfer, eq_self_iff_true, heq_iff_eq, and_self], }
+
+lemma transfer_eq_map_of_le (GH : G ≤ H) :
+  p.transfer H hp = p.map (simple_graph.hom.map_spanning_subgraphs GH) :=
+by { induction p; simp only [*, transfer, map_cons, hom.map_spanning_subgraphs_apply,
+                             eq_self_iff_true, heq_iff_eq, and_self, map_nil], }
+
+@[simp] lemma edges_transfer : (p.transfer H hp).edges = p.edges :=
+by { induction p; simp only [*, transfer, edges_nil, edges_cons, eq_self_iff_true, and_self], }
+
+@[simp] lemma support_transfer : (p.transfer H hp).support = p.support :=
+by { induction p; simp only [*, transfer, eq_self_iff_true, and_self, support_nil, support_cons], }
+
+lemma transfer_is_path (pp : p.is_path) : (p.transfer H hp).is_path :=
+begin
+  induction p;
+  simp only [transfer, is_path.nil, cons_is_path_iff, support_transfer] at pp ⊢,
+  { tauto, },
+end
+
+lemma transfer_is_cycle (p : G.walk u u) (hp) (pc : p.is_cycle) : (p.transfer H hp).is_cycle :=
+begin
+  cases p;
+  simp only [transfer, is_cycle.not_of_nil, cons_is_cycle_iff, transfer, edges_transfer] at pc ⊢,
+  { exact pc, },
+  { refine ⟨transfer_is_path _ _ pc.left, pc.right⟩, },
```

Comment:
It's more customary to use `exact` rather than `refine` when it's finishing off a goal.
```suggestion
  { exact ⟨transfer_is_path _ _ pc.left, pc.right⟩, },
```